File: 프로그래머스/lv2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
def change_number(n, k): # 진수변환
    tmp = ""
    while n > 0:
        tmp += str(n % k)
        n = n // k
    return ''.join(reversed(tmp))

def solution(n, k):
    answer = 0
    k_num = change_number(n, k)
    for n in k_num.split('0'):
        if n == "": continue # 연속되는 0 처리
        if is_prime(int(n)):
            answer += 1
    return answer

def is_prime(n):
    if n == 1: return False
    if n == 2 or n == 3: return True
    tmp = 1
    for i in range(2, int(n**(0.5)+1), tmp):
        if n % i == 0:
            return False
        tmp = i
    return True

File: 프로그래머스/lv2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
import java.util.*;
import java.lang.*;
class Solution {
    public int solution(int n, int k) {
        int answer = 0;
        String tmp = zinsu(n, k);
        String[] lst = tmp.split("0");
        System.out.println(tmp);
        System.out.println(Arrays.toString(lst));
        for (int i =0; i<lst.length; i++){
            if (lst[i].equals("")) continue;
            if (isSosu(Long.parseLong(lst[i]))){
                answer++;
            }
        }
        return answer;
    }
    static String zinsu(int n, int k){
        StringBuilder tmp = new StringBuilder();
        while (n > 0){
            tmp.append(n % k);
            n /= k;
        }
        return tmp.reverse().toString();
    }
    static boolean isSosu(long n){
        if (n < 2) return false;
        if (n == 2) return true;
        for (int i=3; i< Math.sqrt(n)+1; i+=2){
            if (n%i == 0){
                return false;
            }
        }
        return true;
        
    }
}